syntax = "proto3";

package decorcastle;

option java_multiple_files = true;
option go_package = "decorcastle/stock";

// Add options for field naming strategy
option optimize_for = SPEED;

message Stock {
  int32 id = 1;
  int32 product_type_id = 2 [json_name = "productTypeId"];
  optional string product_name = 3 [json_name = "productName"];
  string product_number = 4 [json_name = "productNumber"];
  optional string company = 5; // deprecated, use company_id instead
  int32 company_id = 14 [json_name = "companyId"];
  optional string thickness = 6;
  optional string width = 7;
  optional string height = 8;
  int32 amount_present = 9 [json_name = "amountPresent"];
  optional float unit_price = 10 [json_name = "unitPrice"];
  string created_at = 11 [json_name = "createdAt"];
  string updated_at = 12 [json_name = "updatedAt"];
  optional string last_ordered_at = 13 [json_name = "lastOrderedAt"];
  
  // Adhesive-specific fields
  optional float weight = 15; // Weight in kg for adhesive products
  optional string pack_size = 16 [json_name = "packSize"]; // Pack size description (e.g., '500ml', '1kg')
  
  // Hardware-specific fields
  optional string color = 17; // Color of hardware (e.g., 'Brass', 'Chrome', 'Black')
  optional string alloy_type = 18 [json_name = "alloyType"]; // Alloy type (e.g., 'Stainless Steel', 'Aluminum')
  optional string hardware_category = 19 [json_name = "hardwareCategory"]; // Specific hardware category (e.g., 'wardrobe', 'kitchen', 'general')
}

message CreateStockRequest {
  int32 product_type_id = 1 [json_name = "productTypeId"];
  optional string product_name = 2 [json_name = "productName"];
  string product_number = 3 [json_name = "productNumber"];
  optional string company = 4; // deprecated, use company_id instead
  int32 company_id = 10 [json_name = "companyId"];
  optional string thickness = 5;
  optional string width = 6;
  optional string height = 7;
  int32 amount_present = 8 [json_name = "amountPresent"];
  optional float unit_price = 9 [json_name = "unitPrice"];
  
  // Adhesive-specific fields
  optional float weight = 11; // Weight in kg for adhesive products
  optional string pack_size = 12 [json_name = "packSize"]; // Pack size description (e.g., '500ml', '1kg')
  
  // Hardware-specific fields
  optional string color = 13; // Color of hardware (e.g., 'Brass', 'Chrome', 'Black')
  optional string alloy_type = 14 [json_name = "alloyType"]; // Alloy type (e.g., 'Stainless Steel', 'Aluminum')
  optional string hardware_category = 15 [json_name = "hardwareCategory"]; // Specific hardware category (e.g., 'wardrobe', 'kitchen', 'general')
}

message BatchCreateStockRequest {
  repeated CreateStockRequest items = 1;
}

message UpdateStockRequest {
  optional int32 product_type_id = 1 [json_name = "productTypeId"];
  optional string product_name = 2 [json_name = "productName"];
  optional string product_number = 3 [json_name = "productNumber"];
  optional string company = 4; // deprecated, use company_id instead
  optional int32 company_id = 10 [json_name = "companyId"];
  optional string thickness = 5;
  optional string width = 6;
  optional string height = 7;
  optional int32 amount_present = 8 [json_name = "amountPresent"];
  optional float unit_price = 9 [json_name = "unitPrice"];
  
  // Adhesive-specific fields
  optional float weight = 11; // Weight in kg for adhesive products
  optional string pack_size = 12 [json_name = "packSize"]; // Pack size description (e.g., '500ml', '1kg')
  
  // Hardware-specific fields
  optional string color = 13; // Color of hardware (e.g., 'Brass', 'Chrome', 'Black')
  optional string alloy_type = 14 [json_name = "alloyType"]; // Alloy type (e.g., 'Stainless Steel', 'Aluminum')
  optional string hardware_category = 15 [json_name = "hardwareCategory"]; // Specific hardware category (e.g., 'wardrobe', 'kitchen', 'general')
}
